cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1340,18 +1374,17 @@ private synchronized void startStopProgram(HttpRequest request, HttpResponder re
    * Starts a Program.
    */
   private AppFabricServiceStatus start(final Id.Program id, Map<String, String> overrides, boolean debug) {
-
     try {
-      if (isRunning(id)) {
-        return AppFabricServiceStatus.PROGRAM_ALREADY_RUNNING;
-      }
-
       Map<String, String> sysArgs = propertiesResolver.getSystemProperties(id);
       Map<String, String> userArgs = propertiesResolver.getUserProperties(id);
       if (overrides != null) {
         userArgs.putAll(overrides);
       }
 
+      if (isRunning(id) && isConcurrentRunsDisabled(id, userArgs)) {
```

Comment:
I am not sure I understand this correctly: What are the different ways that the CONCURRENT_RUNS_DISABLED can be set? Is it a property of the program? Or does it have to be set in the preferences store? Or is it a runtime argument? Here it appears that it can only be from preferences, but overridden with runtime arguments? 

I don't know what is the right way to disable/enable it, because I am not sure what the use case is. What would be reason to disable it? And if it is disabled, shouldn't that be a property of the program itself? Or a property of the schedule that triggers it?